docs/automerge.md

```diff
@@ -15,13 +15,24 @@ A pull request will be automerged via a merge commit if it meets the following c
  - Has at least 2 maintainer approvals
  - Base branch is either main or a release branch
 If the pull request has an `automerge` label, the OSM-PR-bot will also autorebase the pull request if the PR branch goes out-of-date.
-> Note: Currently, pull requests that are paths-ignore cannot be merged automatically.
+> Note: Pull requests that are paths-ignore cannot be merged automatically.
+
+A pull request will be autosquashed (commits will be squashed then merged) if it meets the following criteria:
+ - Has the `automerge-squash` label
+ - Does not have the `wip` label
+ - Does not have the `do-not-merge/hold` label
+ - Successfully completed all checks
+ - Has at least 2 maintainer approvals
+ - Base branch is main branch
+ If the pull request has an `automerge-squash` label, the OSM-PR-bot will also autorebase the pull request if the PR branch goes out-of-date.
+> Note: Pull requests that are paths-ignore cannot be merged automatically.
 
 # Autorebase
-A pull request will be autorebased only and not automerged if it has the `autorebase` label. The rebase action will be completed by OSM-PR-bot.
+A pull request will be autorebased only and not automerged/autosquashed if it has the `autorebase` label, as long as there are no conflicts. The rebase action will be completed by OSM-PR-bot.
 
 # In Pull Request Comments
 Some Mergify commands can also be triggered via comments on the pull request.
 - `@Mergifyio refresh` will re-evaluate the rules
 - `@Mergifyio rebase` will rebase this PR on its base branch
+- `@Mergifyio update` will merge the base branch into this PR
```

Comment:
Let's not document this, because it can mess up the Git tree if the merge commit option is used to merge a PR that has multiple commits and when the PR's branch is updated to the base branch using a merge commit.